bbb-common-web/src/main/java/org/bigbluebutton/api/Util.java

```diff
@@ -105,21 +108,42 @@ public String stripPresBaseDirFromPath(String presentationBaseDir, String path)
 		return path;
 	}
 
-	public static File getPresFileDownloadMarker(File presBaseDir, String presId) {
+	public static File getPresFileDownloadMarker(File presBaseDir, String presId, String downloadableExtension) {
 		if (presBaseDir != null) {
-			String downloadMarker = presId.concat(".downloadable");
+			String downloadMarker = presId.concat(".").concat(downloadableExtension).concat(".downloadable");
 			return new File(presBaseDir.getAbsolutePath() + File.separatorChar + downloadMarker);
 		}
 		return null;
 	}
 
+	public static void deleteAllDownloadableMarksInPresentations(File presFileDir, String presId) {
+		String regexString = "\\.(" + String.join("|",
+				SupportedFileTypes.getSupportedFileTypes()) + ")\\.downloadable";
+		File[] filesWithDownloadMark = presFileDir.listFiles(new FilenameFilter() {
+			@Override
+			public boolean accept(File dir, String name) {
+				return name.matches(presId + regexString);
+			}
+		});
+
+		// Iterate through all of downloadable marks to delete them. Pattern is:
+		// <presentationId>.<extension>.downloadable
+		for (File downloadableMark: filesWithDownloadMark) {
+			if (downloadableMark != null && downloadableMark.exists()) {
+				downloadableMark.delete();
+			}
+		}

```

Comment:
```suggestion
		//Delete files that follow the pattern <presentationId>.<extension>.downloadable
		File[] presFiles = presFileDir.listFiles();
		for (File presFile : presFiles) {
			if (presFile.isFile() && presFile.getName().endsWith(".downloadable")) {
				presFile.delete();
			}
		}
```

I suggest a different approach in order to improve readability.